aci/resource_aci_bfdmhifp.go

```diff
@@ -0,0 +1,312 @@
+	if err != nil {
+		return nil, err
+	}
+	bfdMhIfP := models.AciBfdMultihopInterfaceProfileFromContainer(bfdMhIfPCont)
+	if bfdMhIfP.DistinguishedName == "" {
+		return nil, fmt.Errorf("Aci BFD Multihop Interface Profile %s not found", dn)
+	}
+	return bfdMhIfP, nil
+}
+
+func setAciBfdMultihopInterfaceProfileAttributes(bfdMhIfP *models.AciBfdMultihopInterfaceProfile, d *schema.ResourceData) (*schema.ResourceData, error) {
+	dn := d.Id()
+	d.SetId(bfdMhIfP.DistinguishedName)
+	d.Set("description", bfdMhIfP.Description)
+	if dn != bfdMhIfP.DistinguishedName {
+		d.Set("logical_interface_profile_dn", "")
+	}
+	bfdMhIfPMap, err := bfdMhIfP.ToMap()
+	if err != nil {
+		return d, err
+	}
+	d.Set("logical_interface_profile_dn", GetParentDn(dn, fmt.Sprintf("/%s", models.RnbfdMhIfP)))
+	d.Set("annotation", bfdMhIfPMap["annotation"])
+	key := d.Get("key")
+	if key != nil {
+		keyValue := key.(string)
+		if keyValue != "" {
+			d.Set("key", keyValue)
+		}
+	}
+	d.Set("key_id", bfdMhIfPMap["keyId"])
+	d.Set("name", bfdMhIfPMap["name"])
+	d.Set("interface_profile_type", bfdMhIfPMap["type"])
+	d.Set("name_alias", bfdMhIfPMap["nameAlias"])
+	return d, nil
+}
+
+func getAndSetRelationAciBfdMultihopInterfacePolicy(client *client.Client, dn string, d *schema.ResourceData) (*schema.ResourceData, error) {
+	bfdRsMhIfPolData, err := client.ReadRelationbfdRsMhIfPol(dn)
+	if err != nil {
+		log.Printf("[DEBUG] Error while reading relation bfdRsMhIfPol %v", err)
+		d.Set("relation_bfd_rs_mh_if_pol", "")
+		return nil, err
+	} else {
+		d.Set("relation_bfd_rs_mh_if_pol", bfdRsMhIfPolData.(string))
+		log.Printf("[DEBUG] Reading relation bfdRsMhIfPol: %s finished successfully", bfdRsMhIfPolData.(string))
+	}
+	return d, nil
+}
+
+func resourceAciBfdMultihopInterfaceProfileImport(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	log.Printf("[DEBUG] Aci BFD Multihop Interface Profile %s: Beginning Import", d.Id())
+	aciClient := m.(*client.Client)
+	bfdMhIfP, err := getRemoteAciBfdMultihopInterfaceProfile(aciClient, d.Id())
+	if err != nil {
+		return nil, err
+	}
+	schemaFilled, err := setAciBfdMultihopInterfaceProfileAttributes(bfdMhIfP, d)
+	if err != nil {
+		return nil, err
+	}
+
+	getAndSetRelationAciBfdMultihopInterfacePolicy(aciClient, d.Id(), d)
+
+	log.Printf("[DEBUG] %s: Import finished successfully", d.Id())
+	return []*schema.ResourceData{schemaFilled}, nil
+}
+
+func resourceAciBfdMultihopInterfaceProfileCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	log.Printf("[DEBUG] Aci BFD Multihop Interface Profile: Beginning Creation")
+	aciClient := m.(*client.Client)
+	desc := d.Get("description").(string)
+	logicalInterfaceProfileDn := d.Get("logical_interface_profile_dn").(string)
+
+	bfdMhIfPAttr := models.AciBfdMultihopInterfaceProfileAttributes{}
+
+	nameAlias := d.Get("name_alias").(string)
+	if NameAlias, ok := d.GetOk("name_alias"); ok {
+		nameAlias = NameAlias.(string)
+	}
```

Comment:
Would move this around the desc assignment for consistency. Since all the d.Get() methods are called there.

Same applies for the update function and the other resource.

Reply:
changed.